lightning/src/chain/channelmonitor.rs

```diff
@@ -2106,6 +2113,61 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 	}
 }
 
+/// Persist is responsible for persisting channel monitors: this could mean
+/// writing once to disk, and/or uploading to one or more backup services.
+///
+/// Note that for every new monitor, you **must** persist the new ChannelMonitor
+/// to disk/backups. And, on every update, you **must** persist either the
+/// ChannelMonitorUpdate or the updated monitor itself. Otherwise, there is risk
+/// of situations such as revoking a transaction, then crashing before this
+/// revocation can be persisted, then unintentionally broadcasting a revoked
+/// transaction and losing money. This is a risk because previous channel states
+/// are toxic, so it's important that whatever channel state is persisted is
+/// kept up-to-date.
+pub trait Persist<Keys: ChannelKeys>: Send + Sync {
```

Comment:
If it's not bikeshedding, you may move this trait in a new file `persistinterface.rs`. It might be alone for now it should be joined in the future by other persister. IMO, it's better to dissociate clearly the chain monitoring layer from our persistence infra.

Reply:
Do we have specific plans for future persisters yet? Just want to make sure it makes sense to future-proof in this case :)

It seems like how we've been naming traits has been shifting, to be closer to Rust API guidelines (discussion [here](https://github.com/rust-bitcoin/rust-lightning/pull/463#discussion_r370957009)). Jeff suggested this naming, which I think makes sense based on the shift. Persister naming suggestion: https://github.com/rust-bitcoin/rust-lightning/pull/681#discussion_r496465207

Reply:
My two sats:

We should think of traits as defining behaviors (thus 'persist' rather than 'persister'). Traits can be implemented by structs (e.g., a persister). And each such struct may implement multiple traits (i.e., behaviors) associated with persisting many types (e.g., `ChannelMonitor`, `ChannelManager`, etc.).

IMHO, traits for persisting should live with the associated structs. There is no persister infrastructure. Rather, there are structs that may use behaviors for persisting other structs. Thus, `ChainMonitor` knows about persisting `ChannelMonitor` but shouldn't care about persisting `ChannelManager`. So putting all these traits in one module doesn't seem desirable, even though a persister (e.g., `FilesystemPersister`) may need to implement multiple behaviors.